### PR TITLE
fix(server): give IWebSocket.emitToRoom void return type.

### PR DESCRIPTION
**Pull request by WillieHabi**
## Description

I noticed when testing in local server we consistently get this error when disconnecting from document: 

'TypeError: Cannot read properties of undefined (reading 'catch')' 

This is because in disconnectDocument we do `socket.emitToRoom().catch((error) => {...});`

There is an inconsistency in the way we type IWebSocket.emitToRoom. We label it as `async` in some class implementations, but returning a promise is unnecessary given its usage.

https://github.com/microsoft/FluidFramework/blob/17570a472184e0874abf49437e79810414c7670f/server/routerlicious/packages/lambdas/src/nexus/disconnect.ts#L110-L112

This PR gives IWebSocket.emitToRoom a void return type, removes async labels in given implementations, and replaces .catch usage with try/catch blocks instead.

### CODE REVIEW

File: server/routerlicious/packages/local-server/src/localWebSocketServer.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class LocalWebSocket implements IWebSocket {
 		this.events.emit(event, ...args);
 	}
 
-	public emitToRoom(roomId: string, event: string, ...args: any[]) {
+	public async emitToRoom(roomId: string, event: string, ...args: any[]) {
```
Comment (anthony-murphy) [inline server/routerlicious/packages/local-server/src/localWebSocketServer.ts:49]: it looks like the base interface IWebSocket doesn't define a return type for emitToRoom, which defaults to any. so making it a promise also seem incorrect. i think the real fix is to fix the interface, whatever the right return type is, and then ensure all usages are correct as well.

Comment (anthony-murphy) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:49]: on a brief review, i think void is the correct return type, and the assumption that it should be a promise is incorrect

Comment (WillieHabi) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:49]: To reconcile the usage of emitToRoom.catch() ([here](https://github.com/microsoft/FluidFramework/blob/054480f1868dfb91c3b9270d11aaae9f5533068d/server/routerlicious/packages/lambdas/src/nexus/disconnect.ts#L110C1-L112C23) and [here](https://github.com/microsoft/FluidFramework/blob/054480f1868dfb91c3b9270d11aaae9f5533068d/server/routerlicious/packages/lambdas/src/nexus/connect.ts#L430C1-L432C31)), would it be better to change these to try/catch instead?

Comment (anthony-murphy) [reply server/routerlicious/packages/local-server/src/localWebSocketServer.ts:49]: if it turns out that IWebSocket.emitToRoom can have a void return, specifically that there is not an implementation that needs it to be a promise, then yes, they should move to try catch. 

Right now the usage is inconsistent, some places, like those you call out assume it's a promise, whereas other places assume it's not. Adding consistency is the most important things here, which means explicitly typing the return value on the interface. 

It seems the right return type is either void or promise<void>. I would prefer void unless there is an implementation of the interface that needs it to be a promise, which i didn't see when i briefly reviewed. Once the interface is well defined, you will then need to make the usages consistent, but this should be easy as there should be compile or lint errors where it is wrong

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**